samples/drivers/adc/adc_dt/boards/xg29_rb4412a.overlay

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ */
+
+#include <dt-bindings/pinctrl/silabs/xg29-pinctrl.h>
+#include <dt-bindings/adc/silabs-adc.h>
+
+/ {
+	zephyr,user {
+		io-channels = <&adc0 3>, <&adc0 4>;
+	};
+};
+
+&pinctrl {
+	adc0_default: adc0_default {
+		group0 {
+			/* Allocate odd bus 0 on GPIO port B to IADC for access to pin PB1 */
+			silabs,analog-bus = <ABUS_BODD0_IADC0>;
+		};
+	};
+};
+
+&adc0 {
+	pinctrl-0 = <&adc0_default>;
+	pinctrl-names = "default";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	channel@3 {
+		reg = <3>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+		zephyr,resolution = <12>;
+		zephyr,input-positive = <IADC_INPUT_AVDD>;
+	};
+
+	channel@4 {
+		reg = <4>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_VDD_1";
+		zephyr,vref-mv = <3300>;
+		zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+		zephyr,resolution = <12>;
+		zephyr,input-positive = <IADC_INPUT_PB1>;
```

Comment:
Should this sample use PD3 instead? It is connected to the joystick, which allows for multiple analog measurements, as opposed to PB1, which is connected to a button that only has VDD and GND states. The connection to the joystick isn't present on xg24_rb4187c (0 ohm resistor not mounted), which is why it uses a button.